data/tmdb-shows/src/main/java/io/filmtime/data/tmdb/shows/TmdbShowsRepositoryImpl.kt

```diff
@@ -73,4 +79,17 @@ internal class TmdbShowsRepositoryImpl @Inject constructor(
 
   override suspend fun episodesBySeason(showId: Int, seasonNumber: Int): Result<List<EpisodeThumbnail>, GeneralError> =
     tmdbShowsRemoteSource.episodesBySeason(showId, seasonNumber)
+
+  override suspend fun getBookmarkedShows(): Flow<List<VideoThumbnail>> {
+    return bookmarksDao.getAllBookmarksByType(Show).map { bookmarks ->
+      val showsList = mutableListOf<VideoThumbnail>()
+      for (bookmark in bookmarks) {
+        when (val detail = showDetails(bookmark.tmdbId)) {
+          is Failure -> continue
+          is Success -> showsList.add(detail.data.toVideoThumbnail())
+        }
+      }
+      showsList
```

Comment:
Not a good idea to make multiple requests every time user asks for bookmarks. Maybe saving thumbnail image helps here. I'll create an issue for it #119